### ISSUE TITLE
Support concatenating programs

### ISSUE THREAD

**Issue by notmgsk** [rigetti/quil-rust#65]
I recently had to build a program from two different program sources: one with the calibrations, and another with the actual program logic. It was not exactly ergonomic, and I think we should support something akin to pyQuil's program concatenation.

**Comment by kris524** [comment]
Hi @notmgsk, Is this issue still relevant? I can try and help wherever possible.

**Comment by notmgsk** [comment]
@kris524 Yeah! That would be great. Take a look at how pyQuil does the same. There is some care to be taken, especially WRT combining calibrations, frames, etc. from both programs. Let me know if you have more questions.

**Comment by kris524** [comment]
Hi @notmgsk, I had a look at some files but I have difficulty finding the ones you mention. Do you know the exact paths to the files or examples I can have a look at? I can find the `calibration.rs` but not sure where to look in pyQuil

**Comment by erichulburd** [comment]
@jselig-rigetti It seems to me this is already resolved via `impl From<Vec<Instruction>> for Program` and `impl ops::Add<Program> for Program`, right? I'm going to close this, but feel free to re-open if I'm missing something.
